packages/backend/src/studio.ts

```diff
@@ -213,6 +213,7 @@ export class Studio {
      */
     this.#catalogManager = new CatalogManager(this.#rpcExtension, appUserDirectory);
     this.#catalogManager.init();
+    this.#extensionContext.subscriptions.push(this.#catalogManager);
```

Comment:
This was missing, it's my understanding that it should be pushed to the subscriptions for disposal.

Reply:
It is now optional as every Disposable it automatically disposed when extension is stopped. Should be moved to a separate PR

Reply:
I've removed the _missing_ entries.
However, I'm having a hard time understanding how are the `Disposable`s discovered by the extension for automatic disposal.

Reply:
Yeah, for extension-defined disposable, it should be disposed by the extension. The disposable provided by the extension-api are disposed by the core, but I think it is always a good idea to do it, for good practice.